packages/terra-action-header/tests/jest/ActionHeader.test.jsx

```diff
@@ -91,4 +91,28 @@ describe('ActionHeader', () => {
     const wrapper = shallowWithIntl(actionHeader);
     expect(wrapper).toMatchSnapshot();
   });
+
+  it('correctly applies the theme context className', () => {
+    jest.spyOn(React, 'useContext')
+      .mockReturnValue({
+        className: 'orion-fusion-theme',
+      });
+
+    const actionHeader = (
+      <ActionHeader
+        title="Action Header"
+        className="customClassName"
+        onMaximize={() => {}}
+        onClose={() => {}}
+      />
+    );
+    const wrapper = shallowWithIntl(actionHeader);
+    const headerContainer = wrapper.dive();
+
+    expect(headerContainer.find('.flex-header').hasClass('orion-fusion-theme')).toBeTruthy();
+    expect(headerContainer.find('.left-buttons').hasClass('orion-fusion-theme')).toBeTruthy();
+    expect(headerContainer.find('.right-buttons').hasClass('orion-fusion-theme')).toBeTruthy();
```

Comment:
IMO this is overkill. As long as the snapshot shows the orion-fusion-theme class got added to the top level div in the snapshot you don't have to test that the DOM appropriately applies classes to child nodes.